File: ll/LinkedList.java

public class LinkedList {
    private int val;
    private LinkedList next;

    public LinkedList (int val, LinkedList next) {
        this.val = val;
        this.next = next;
    }

    public LinkedList (int val) {
        this(val, null);
    }

    public void print () {
        if (next != null) {
            System.out.print(val + " -> ");
            next.print();
        } else {
            System.out.println(val);
        }
    }

    public void print_reverse() {
        if (next != null) {
            next.print_reverse();
        } 
        System.out.print(val + " -> ");
    }

    public void insert_end (int val) {
        if (next == null) {
            next = new LinkedList(val);
        } else {
            next.insert_end(val);
        }
    }

    public static void main(String[] args) {
        LinkedList head = new LinkedList(10);
        for (int i = 0; i < 10; i ++)
            head.insert_end(i);
        head.print();
        // head.print_reverse();
    }
}

File: ll/linkedList.py
class LinkedList:
    def __init__(self, val, next_node=None):
        self.val = val
        self.next = next_node
    
    def print (self):
        if self is None:
            return

        if self.next is not None:
            print(self.val, "->", end=" ")
            self.next.print()
        else:
            print(self.val)
    
    def insert_end (self, val):
        if self.next is None:
            self.next = LinkedList(val,None)
        else:
            self.next.insert_end(val)

if __name__ == "__main__":
    head = LinkedList(5)
    head.insert_end("Wot")
    head.insert_end(8)
    for i in range(0,10):
        head.insert_end(i)
    head.print()